### PR TITLE
Add SPV_INTEL_optnone extension spec

### PR DESCRIPTION
**Pull request by Fznamznon**

### CODE REVIEW

**Comment by Fznamznon** [comment]
Something is wrong with codeowners. I believe I was needed review from @intel/dpcpp-specification-reviewers .

**Comment by dm-vodopyanov** [comment]
@Fznamznon , +1, CODEOWERS are the following:
```sycl/doc/extensions/ @intel/dpcpp-specification-reviewers```
Please ping me or someone else from intel/llvm-reviewers-runtime to approve this after the approval from dpcpp-specification-reviewers to unblock the merging.

**Comment by bader** [comment]
> Something is wrong with codeowners. I believe I need review from @intel/dpcpp-specification-reviewers .

Fixed.

File: sycl/doc/extensions/SPIRV/SPV_INTEL_optnone.asciidoc
Comment (AlexeySachkov) [inline sycl/doc/extensions/SPIRV/SPV_INTEL_optnone.asciidoc:14]:
```suggestion
https://github.com/intel/llvm
```

File: sycl/doc/extensions/SPIRV/SPV_INTEL_optnone.asciidoc
Comment (AlexeySachkov) [inline sycl/doc/extensions/SPIRV/SPV_INTEL_optnone.asciidoc:54]:
```suggestion
This extension adds new possible value for Function Control mask - *OptNoneINTEL*,
which represents a strong request to not optimize the function.
```

**Review by AlexeySachkov — APPROVED** [review]

File: sycl/doc/extensions/SPIRV/SPV_INTEL_optnone.asciidoc
Comment (Fznamznon) [reply sycl/doc/extensions/SPIRV/SPV_INTEL_optnone.asciidoc:14]: Why?

Comment (AlexeySachkov) [reply sycl/doc/extensions/SPIRV/SPV_INTEL_optnone.asciidoc:14]: We are not publishing it there so I don't understand why do we need to redirect questions there

File: sycl/doc/extensions/SPIRV/SPV_INTEL_optnone.asciidoc
Comment (AlexeySotkin) [inline sycl/doc/extensions/SPIRV/SPV_INTEL_optnone.asciidoc:75]: The table in section 3.31 has slightly different header now.

```suggestion
2+^| Capability ^| Implicitly Declares
```

**Review by AlexeySotkin — APPROVED** [review]

**Comment by AlexeySotkin** [comment]
BTW, do we really need to run tests for doc-only patches?

**Comment by AlexeySotkin** [comment]
@intel/dpcpp-specification-reviewers please review.

**Comment by dm-vodopyanov** [comment]
@Fznamznon, is this PR still relevant? Should it be merged?

**Comment by Fznamznon** [comment]
> @Fznamznon, is this PR still relevant? Should it be merged?

I don't see why not. It is just yet another SPIR-V spec that wasn't merged because we didn't have dedicated SPIR-V spec codeowners back then.

**Pull request merged**
